edb/pgsql/compiler/pathctx.py

```diff
@@ -124,6 +133,7 @@ def get_path_var(
             src_path_id = path_id
         else:
             src_path_id = path_id.src_path()
+            assert src_path_id is not None
```

Comment:
This is subtle and I can't see. I looked in ir/pathid.py and in the src_path() method but I don't quite get it. Why do we know this will never be None?

Reply:
It is correlated with `path_id.rptr()`, which we check above.  If `path_id.rptr()` is not `None`, then `src_path()` would not be.  Both are essentially `[:-1]` slices on the path id.